File: mypay.sources/mypay4-be/src/main/java/it/regioneveneto/mygov/payment/mypay4/model/EnteFunzionalita.java
/**
 *     MyPay - Payment portal of Regione Veneto.
 *     Copyright (C) 2022  Regione Veneto
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package it.regioneveneto.mygov.payment.mypay4.model;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "mygovEnteFunzionalitaId")
public class EnteFunzionalita extends BaseEntity {

  public final static String ALIAS = "EnteFunzionalita";
  public final static String FIELDS = ""+ALIAS+".mygov_ente_funzionalita_id as EnteFunzionalita_mygovEnteFunzionalitaId"+
      ","+ALIAS+".cod_ipa_ente as EnteFunzionalita_codIpaEnte,"+ALIAS+".cod_funzionalita as EnteFunzionalita_codFunzionalita"+
      ","+ALIAS+".flg_attivo as EnteFunzionalita_flgAttivo";

  private Long mygovEnteFunzionalitaId;
  private String codIpaEnte;
  private String codFunzionalita;
  private boolean flgAttivo;

  //calculated fields
  private Timestamp dtUltimaAbilitazione;
  private Timestamp dtUltimaDisabilitazione;

}

File: mypay.sources/mypay4-fe/projects/mypay4-fe-operatore/src/app/model/ente-funzionalita.ts
/*
 *     MyPay - Payment portal of Regione Veneto.
 *     Copyright (C) 2022  Regione Veneto
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { DateTime } from 'luxon';
import { MapperDef, MapperType, WithActions } from 'projects/mypay4-fe-common/src/public-api';

export class EnteFunzionalita extends WithActions {
  public static readonly MAPPER_S2C_DEF = [
    new MapperDef(MapperType.Function,'statoAbilitazione',(enteFunzionalita: EnteFunzionalita) => {
      return enteFunzionalita.flgAttivo ? 'Abilitato' : 'Disabilitato';
    }),
    new MapperDef(MapperType.DateTime,'dtUltimaAbilitazione','local-date-time'),
    new MapperDef(MapperType.DateTime,'dtUltimaDisabilitazione','local-date-time'),
  ]

  mygovEnteFunzionalitaId: number;
  codIpaEnte: string;
  codFunzionalita: string;
  flgAttivo: boolean;
  dtUltimaAbilitazione: DateTime;
  dtUltimaDisabilitazione: DateTime;

  statoAbilitazione: string;
}
